pandas/core/arrays/arrow/array.py

```diff
@@ -1012,7 +1012,11 @@ def _concat_same_type(
         ArrowExtensionArray
         """
         chunks = [array for ea in to_concat for array in ea._data.iterchunks()]
-        arr = pa.chunked_array(chunks)
+        if to_concat[0].dtype == "string":
```

Comment:
Is needed specifically for `StringDtype("pyarrow")` and not `ArrowDtype(pa.string())`?

If so, could you add a comment to that effect?

Reply:
Yes since the StringDtype does not have a pyarrow_dtype

Reply:
Added